File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.c


int sumOddLengthSubarrays(int* arr, int arrSize){
    

    int size = 1;
    int sum = 0;
    
    while(size <=arrSize){
        for(int i = 0; i < arrSize;i++){
            for(int j = i; j < size+i && size+i <= arrSize;j++){
                sum += arr[j];
            }
        }
        size += 2;
    }
    return sum;
}

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int total = 0;
        for (int i = 0; i < arr.length;i+=1){
            total += ((i+1)*(arr.length-i)+1)/2 * arr[i];
        }
        return total;      
    }
}